Assets/Scripts/NPCs/s_Manager_NPC.cs

```diff
@@ -0,0 +1,151 @@
+
+            //Branching action based on the time until it's allowed to walk again.
+            if (currentWaitTime != 0)  //If the time is not zero.
+            {
+                currentWaitTime--;  //Decrement the time by 1.
+                CastCircleCast();   //And cast the circlular Raycast.
+            }
+            else //But if the time is zero.
+            {
+                currentWaitTime = maxWaitTIme; //reset the timer that it must wait.
+                ChangeDestination();    //Set a new destination. This could end up being the same point that it is on, which case it will re-roll on the next frame.
+                                        //If the destination is at a different location than the current position, the NPC will walk.
+                                        //If not, it's just waiting at the same spot and re-reolls a new destination.
+                                            //This has the addition of acting like a second layer of rng. It could stay because the location is repeatedly the same, or it moves.
+
+            }
+
+        }
+
+    }
+
+    void Walk()
+    {
+        if (transform.position.x < pausePoints[currentTargetPoint].transform.position.x) //NPC is left of destination, moving right
+        {
+            transform.Translate((Vector3.right) * Time.deltaTime * walkSpeed);  //Move to new location right.
+            thisSpriteRenderer.flipX = true;  //Face sprite RIght
+
+        }
+        else //NPC is right of destination, moving left.
+        {
+            transform.Translate((-Vector3.right) * Time.deltaTime * walkSpeed); //Move to the location left.
+            thisSpriteRenderer.flipX = false; //Face Sprite Left
+
+        }
+    }
+
+    void ChangeDestination()
+    {
+        int oldTargetPoint = currentTargetPoint; //Copy current target number to oldTarget
+        currentTargetPoint = Random.Range(0, pausePoints.Length);   //set current target to a new number.
+
+        if(oldTargetPoint != currentTargetPoint && punishmentEnabled == false) //if the old number and the new number are different & ppunishment is disabled
+        {
+            punishmentEnabled = true;   //Then re-enable raycast.
+        }
+    }
+
+    void CastCircleCast()
+    {
+        RaycastHit2D[] hits = Physics2D.CircleCastAll(transform2D, circleCastRadius, transform.right, LayerMask.NameToLayer("Player")); //Add player layer later.
+
+        foreach (RaycastHit2D hit in hits) //This can be removed later. Simply used to tell me if something is being detected.
+        {
+            Debug.Log("Hit: " + hit.collider.name + "\n");
+        }
+
+        //Real detection code.
+        if(hits.Length > 0) //If the player is detected, they are in view, and may now be punished if they hit the Duct Tape Button.
+        {
+
+            if(Input.GetKeyDown(KeyCode.Space) && punishmentEnabled == true){ //Replace "Input.GetKeyDown(KeyCode.Space)" with a call to the player controller.
```

Comment:
This code is using the old input system

Reply:
yes, "Input.GetKeyDown(KeyCode.Space" will instead be a reference to the player controller, so that the script can detect if that button has been pressed. So that will change once I've got the real player controller.